spec/javascripts/govuk_publishing_components/load-analytics.spec.js

```diff
@@ -0,0 +1,323 @@
+  })
+
+  describe('for GA4', function () {
+    beforeEach(function () {
+      window.GOVUK.analyticsGa4.vars = null
+      expected = {
+        id: 'GTM-MG7HG5W',
+        auth: undefined,
+        preview: undefined,
+        environment: 'production',
+        gem_version: 'not found'
+      }
+    })
+
+    it('loads GA4 on production domains', function () {
+      var domains = [
+        'www.gov.uk',
+        'www-origin.publishing.service.gov.uk'
+      ]
+
+      for (var i = 0; i < domains.length; i++) {
+        window.GOVUK.analyticsGa4.vars = null
+        var domain = domains[i]
+        window.GOVUK.loadAnalytics.loadGa4(domain)
+        expect(window.GOVUK.analyticsGa4.vars.environment).toEqual('production')
+      }
+    })
+
+    it('loads GA4 on staging domains', function () {
+      var domains = [
+        'www.staging.publishing.service.gov.uk',
+        'www.staging.publishing.service.gov.uk',
+        'www-origin.staging.publishing.service.gov.uk'
+      ]
+      for (var i = 0; i < domains.length; i++) {
+        window.GOVUK.analyticsGa4.vars = null
+        var domain = domains[i]
+        window.GOVUK.loadAnalytics.loadGa4(domain)
+        expect(window.GOVUK.analyticsGa4.vars.environment).toEqual('staging')
+      }
+    })
+
+    it('loads GA4 on integration domains', function () {
+      var domains = [
+        'www.integration.publishing.service.gov.uk',
+        'www.integration.publishing.service.gov.uk',
+        'www-origin.integration.publishing.service.gov.uk'
+      ]
+
+      for (var i = 0; i < domains.length; i++) {
+        window.GOVUK.analyticsGa4.vars = null
+        var domain = domains[i]
+        window.GOVUK.loadAnalytics.loadGa4(domain)
+        expect(window.GOVUK.analyticsGa4.vars.environment).toEqual('integration')
+      }
+    })
+
+    it('loads GA4 on development domains', function () {
+      var domains = [
+        'localhost', '127.0.0.1', '0.0.0.0', 'static.dev.gov.uk'
+      ]
+
+      for (var i = 0; i < domains.length; i++) {
+        window.GOVUK.analyticsGa4.vars = null
+        var domain = domains[i]
+        window.GOVUK.loadAnalytics.loadGa4(domain)
+        expect(window.GOVUK.analyticsGa4.vars.environment).toEqual('development')
+      }
+    })
+
+    it('ignores loading GA4 on unrecognised domains', function () {
+      var domains = [
+        'www.nationalarchives.gov.uk'
+      ]
+
+      for (var i = 0; i < domains.length; i++) {
+        window.GOVUK.analyticsGa4.vars = null
+        var domain = domains[i]
+        window.GOVUK.loadAnalytics.loadGa4(domain)
+        expect(window.GOVUK.analyticsGa4.vars).toEqual(null)
+      }
+    })
+
+    it('sets the correct production variables', function () {
+      window.GOVUK.loadAnalytics.loadGa4('www.gov.uk')
+      delete window.GOVUK.analyticsGa4.vars.internalDomains
+      expect(window.GOVUK.analyticsGa4.vars).toEqual(expected)
+    })
+
+    it('sets the correct staging variables', function () {
```

Comment:
These tests and the ones above (`loads GA4 on x domains`) seem to call the same bit of code, would it be worth combining them or is there a reason they're separated out?

Reply:
@andysellick I think it's a bit more readable/clean having them in separate functions, especially because when the individual test fails you know exactly which part of the code is failing. Do you think it's worth combining them?

Reply:
No, that makes sense, thanks.